app/src/signing/networkMessages/polkadot.js

```diff
@@ -154,6 +154,26 @@ export async function ClaimRewardsTx(senderAddress, {}, network, api) {
   return await getSignMessage(senderAddress, allClaimingTxs, api)
 }
 
+// Vote
+export async function VoteTx(
+  senderAddress,
+  { proposalId, lockedBalance, voteOption, conviction },
+  network,
+  api
+) {
+  const referendumId = proposalId.split("-")[1]
```

Comment:
hack, can we get this without string manipulation?

Reply:
not here, not in the FE. I think doing this is fine. I can add a comment in API though highlighting the importance of being consistent with this format

Reply:
how about sending the actual id from the API?

Reply:
yeah, that would work

Reply:
Then we should add a new field in type Proposal too: `proposalType` , that would be an enum with `referemdum`, `democracy` and so on, so we can now in the FE which one to display as id.

But wait, now thinking about it, it is not a good solution, since we need different ids because of cache... otherwise we will get cache errors using the same id (same number)

Reply:
you can't use the property `id`. it would be interpreted as a cachable id. you need to use sth like `proposalId` or `onChainId`

Reply:
Done